test/langtools/tools/javac/options/BCPOrSystemNotSpecified.java

```diff
@@ -206,7 +209,9 @@ private void prepareBCP(Path target) throws IOException {
                     Files.createDirectories(targetDir);
                     try (InputStream in = file.openInputStream()) {
                         String sourcePath = file.getName();
-                        int sepPos = sourcePath.lastIndexOf(fileSep);
+                        // Here, we should use file system separator instead of the operating system separator.
+                        String fileSystemSep = ((PathFileObject) file).getPath().getFileSystem().getSeparator();
```

Comment:
It should be possible to get the `Path` without having to cast to the internal `PathFileObject` type.

See https://docs.oracle.com/en/java/javase/15/docs/api/java.compiler/javax/tools/StandardJavaFileManager.html#asPath(javax.tools.FileObject)

Reply:
@jonathan-gibbons Thanks for your suggestion. I have updated the code.